File: nodejs/model/market.ts
import { CommonThing } from 'common.ts';

export type TradeGood = CommonThing & {
	activity: string;
	purchasePrice: number;
	sellPrice: number;
	supply: string;
	tradeVolume: number;
	type: string;
};

export type Market = {
	symbol: string;
	exchange: Array<CommonThing>;
	exports: Array<CommonThing>;
	imports: Array<CommonThing>;
	//transactions: Array<Transaction>;
	tradeGoods: Array<TradeGood>;
};

File: golang/pkg/model/market.go
package model

type Market struct {
	Exchange     []Common      `json:"exchange"`
	Exports      []Common      `json:"exports"`
	Imports      []Common      `json:"imports"`
	Symbol       string        `json:"symbol"`
	TradeGoods   []TradeGood   `json:"tradeGoods"`
	Transactions []Transaction `json:"transactions"`
}
